datastore/src/main/java/org/jboss/pnc/datastore/repositories/internal/AbstractRepository.java

```diff
@@ -33,20 +34,30 @@
 
     protected JpaRepository<T, ID> springRepository;
     protected JpaSpecificationExecutor<T> springSpecificationsExecutor;
+    protected EntityManager entityManager;
```

Comment:
Why not use just this instead of passing it through constructor?
```suggestion
    @Inject
    protected EntityManager entityManager;
```

Reply:
Can it be used in the abstract class? I was not sure about it. I thought it is not possible. If it is possible, then yes, it is much better.

Reply:
it is possible. I use it like this in abstratProvider I think

Reply:
Done